packages/astro/src/core/app/index.ts

```diff
@@ -51,6 +80,15 @@ export interface RenderErrorOptions {
 }
 
 export class App {
+
+	/**
+	 * Symbols that the Astro app reads on the passed Request instance. Use these when you can't directly provide these values to `app.render()`.
+	 */
+	static readonly Symbol = Object.freeze({
+		locals: Symbol.for('astro.locals'),
+		clientAddress: Symbol.for('astro.clientAddress'),
+	})
```

Comment:
If we want to **expose** these symbols to user-land, I think we should do it a stricter way. We have been using `Symbol.for` because we know that this is internal information.

Although, this API adds a layer fo contract with the adapters. I think it's time to use  e.g.`Symbol("astro.locals")` and use this one symbol across the board. 

Also, I would expose these values via a getter function:

```js
import { LocalsSymbol } from "../constants.js"

class App {
	get localsSymbol() {
		return LocalsSymbol;
	}
}
```

Reply:
I am guessing the getter is for preventing mutation. Object.freeze should have us covered there.

Reply:
> I think it makes sense to keep the symbol registered for compatibility. Adapters may also have constraints that prevent them from importing anything in the relevant code path, using the registered symbol directly could help there.

I can't understand your point here.

> Exposing it here serves the purpose of not having to remember the casing etc ("did it have dots or underscores?")

And that's why we should have the casing written in one single place :)

Feel free to keep using `Symbol.for`, but the actual string now should be saved somewhere as constant and shared.

Reply:
Again, I don't understand why this is an issue. An adapter is already doing `import { App } from "astro/app"`, so it's importing code from astro

Reply:
That is true, it is for cases where the adapter may want to load locals in another file which does not import `App`, and can't. Basically, `Symbol.for` is versatile.

Reply:
> And that's why we should have the casing written in one single place :)

I agree. After having looked at the our use of registered symbols, it's widespread enough to warrant its own PR.